components/web/ejabberd/Makefile

```diff
@@ -21,60 +21,80 @@
+# Copyright 2017 Andreas Grueninger, Grueninger GmbH, (grueni). All rights reserved. 
 #
 
 include ../../../make-rules/shared-macros.mk
 
-# for ginstall and tail -n
-PATH=/usr/gnu/bin:/usr/bin
-
-COMPONENT_NAME=		ejabberd
-COMPONENT_VERSION=	2.1.13
+COMPONENT_NAME= ejabberd
+COMPONENT_FMRI= web/server/$(COMPONENT_NAME)
+COMPONENT_VERSION= 17.11
 COMPONENT_PROJECT_URL=	http://www.ejabberd.im/
-COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
-COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tgz
-COMPONENT_ARCHIVE_HASH=	\
-    sha256:a80d1c72e435da31758261cb0c04fe750ac71fc3e33a3b606123efd3b3abc86d
-COMPONENT_ARCHIVE_URL=	http://www.process-one.net/downloads/ejabberd/$(COMPONENT_VERSION)/$(COMPONENT_ARCHIVE)
-COMPONENT_BUGDB=	service/ejabberd
-
-include ../../../make-rules/prep.mk
-
-SRC_OFFSET =	src
-# Now we override paths to .built
-BUILD_DIR_32 = $(BUILD_DIR)/$(MACH32)/$(SRC_OFFSET)
-
-COMPONENT_PREP_ACTION = \
-       (cd $(@D)/src ; PATH=$(PATH) $(AUTORECONF) -fiv)
-
-
-include $(WS_TOP)/make-rules/configure.mk
-include $(WS_TOP)/make-rules/ips.mk
+COMPONENT_SRC= $(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE= $(COMPONENT_SRC).tgz
+COMPONENT_ARCHIVE_HASH= sha256:bd16b33a2f1345e5847fd05b70a5fa992181bac756cd6d5d621de9c18cb76801
+COMPONENT_ARCHIVE_URL= https://www.process-one.net/downloads/ejabberd/$(COMPONENT_VERSION)/$(COMPONENT_ARCHIVE)
+COMPONENT_BUGDB= service/ejabberd
+COMPONENT_SUMMARY= ejabberd - Jabber/XMPP instant messaging server
+COMPONENT_DESCRIPTION= the ejabberd Jabber/XMPP instant messaging server
+COMPONENT_CLASSIFICATION= Web Services/Communications
+COMPONENT_LICENSE= GPL-2.0
+COMPONENT_LICENSE_FILE= $(COMPONENT_NAME).license
+
+include $(WS_MAKE_RULES)/prep.mk
+include $(WS_MAKE_RULES)/configure.mk
+include $(WS_MAKE_RULES)/ips.mk
+
+CFLAGS   += $(JPEG_CPPFLAGS) $(JPEG_CFLAGS) -I$(USRINCDIR)/gd2
+CXXFLAGS += $(JPEG_CPPFLAGS) $(JPEG_CXXFLAGS) -I$(USRINCDIR)/gd2
+CPPFLAGS += $(JPEG_CPPFLAGS) -I$(USRINCDIR)/gd2
+LDFLAGS  += $(JPEG_LDFLAGS) -lnsl -lsocket
 
-COMPONENT_BUILD_ENV += "HOME=/tmp"
 CONFIGURE_ENV += "HOME=/tmp"
+CONFIGURE_ENV += "CPP=$(CC) -E"
+CONFIGURE_ENV += "CXX=$(CXX) $(CC_BITS)"
+COMPONENT_BUILD_ENV += $(CONFIGURE_ENV)
 COMPONENT_INSTALL_ENV += "HOME=/tmp"
 
-CONFIGURE_SCRIPT = $(SOURCE_DIR)/src/configure
-COMPONENT_PRE_CONFIGURE_ACTION = \
-	($(CLONEY) $(SOURCE_DIR) $(BUILD_DIR)/$(MACH32))
-
-# tail with -n is needed
-CONFIGURE_ENV += "PATH=/usr/xpg4/bin:/usr/bin"
+COMPONENT_PRE_BUILD_ACTION+= \
+  $(CLONEY) $(SOURCE_DIR) $(@D)
 
 CONFIGURE_OPTIONS +=	--sysconfdir=$(ETCDIR)
 CONFIGURE_OPTIONS +=	--with-erlang=$(USRBINDIR)
 CONFIGURE_OPTIONS +=	--localstatedir=/var
 CONFIGURE_OPTIONS +=	--enable-odbc
+CONFIGURE_OPTIONS +=	--enable-mysql
+CONFIGURE_OPTIONS +=	--enable-pgsql
+CONFIGURE_OPTIONS +=	--enable-sqlite
+CONFIGURE_OPTIONS +=	--enable-pam
+CONFIGURE_OPTIONS +=	--enable-riak
+CONFIGURE_OPTIONS +=	--enable-redis
+CONFIGURE_OPTIONS +=	--enable-user=daemon
```

Comment:
When this option is set, gmake install fails, because it tries to chown files to daemon.

Reply:
I removed the option.
I thought user daemon exists in each system.

Reply:
The issue is not that it doesn't exist. The issue is that gmake install is run as usual user, but tries to change files owner.